ui/apps/platform/cypress/integration/networkGraph/networkGraph.test.js

```diff
@@ -66,4 +69,28 @@ describe('Network Graph Search', () => {
             expect(minDeps).to.include.members(['central', 'scanner', 'sensor']);
         });
     });
+
+    it('should render an error message when the server fails to return a successful response', () => {
+        visitNetworkGraph();
+
+        // Stub out an error response from the server
+        const error =
+            'Number of deployments (2200) exceeds maximum allowed for Network Graph: 2000';
+        cy.intercept('GET', api.network.networkGraph, {
+            statusCode: 500,
+            body: { error, message: error },
+        }).as('networkGraph');
+        selectNamespaceFilters('stackrox', 'kube-system');
+        cy.wait('@networkGraph');
+
+        cy.get(networkPageSelectors.errorOverlay.heading);
+        cy.get(networkPageSelectors.errorOverlay.message(error));
+
+        // Ignore previously stubbed error response and allow the request to respond normally
+        cy.intercept('GET', api.network.networkGraph, (req) => req.continue()).as('networkGraph');
+        selectNamespaceFilters('kube-system');
+        cy.wait('@networkGraph');
+
+        cy.get(networkPageSelectors.errorOverlay.heading).should('not.exist');
```

Comment:
To balance a negative assertion with a positive assertion, does it work to add:

```js
cy.get(networkPageSelectors.cytospaceContainer);
```

Reply:
Yes, good call!